src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -286,11 +286,11 @@ impl<'a, 'cl> Resolver<'a, 'cl> {
                         // while the current crate doesn't have a valid `crate_name`.
                         if crate_name != keywords::Invalid.name() {
                             // `crate_name` should not be interpreted as relative.
-                            module_path.push(Ident {
+                            module_path.push((Ident {
                                 name: keywords::CrateRoot.name(),
-                                span: source.span,
-                            });
-                            source.name = crate_name;
+                                span: source.0.span,
+                            }, Some(ast::CRATE_NODE_ID)));
```

Comment:
This should be a fresh node id, I think, but the id that is getting multiply resolved is not `CRATE_NODE_ID`